### PR TITLE
New: `one-var-declaration-per-line` rule (fixes #1622)

### LINKED ISSUE #1622: missing rule: require line break after variable assignment
Hi. I've tried to review each and every rule of JSHint, JSCS and ESLint to see if I could entirely "get rid of" the first two in favour of the latter one. As suggested by the owner of the @geteslint twitter account, I'm reporting the JSCS rules I've found actually not covered by ESLint. My apologies if it's a duplicate.

- "requireLineBreakAfterVariableAssignment"

E.g this is invalid

```js
var abc = 8; var foo = 5;

var a, b, c,
    foo = 7, bar = 8;
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/7466067-missing-rule-require-line-break-after-variable-assignment?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: docs/rules/one-var-declaration-per-line.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+## Rule Details
+
+This rule enforces a consistent coding style where newlines are required or disallowed around each var declaration inside a variable declaration statement or just when there is a variable initialization. It ignores var declarations inside for loop conditionals.
+
+### Options
+
+This rule takes one option, a string, which can be:
+
+* `"all"` enforces a newline after each variable declaration (default)
```
Comment (nzakas) [inline docs/rules/one-var-declaration-per-line.md]: Maybe this should be "always", so we can also have "never"? (I'm sure someone will ask for that.)

Comment (alberto) [inline docs/rules/one-var-declaration-per-line.md]: I avoided using "always" because "never" didn't make sense to me.

Comment (nzakas) [inline docs/rules/one-var-declaration-per-line.md]: "Never" would mean you must do this:

```js
var a, b, c=0;
```

I think it's a logical opposite of "always" with "initializations" being a middle ground.

Comment (alberto) [inline docs/rules/one-var-declaration-per-line.md]: PR updated.

**Comment by nzakas** [comment]
Lgtm. Really nice job on this.

**Comment by alberto** [comment]
Thanks :)

File: docs/rules/one-var-declaration-per-line.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Require or disallow an newline around variable declarations (one-var-declaration-per-line)
+
+Some developers declare multiple var statements on the same line:
+
+```js
+var foo, bar, baz = 0;
```
Comment (mjomble) [inline docs/rules/one-var-declaration-per-line.md]: The descriptions below sound like this rule can be used to enforce this style as well.
However, the examples for both "always" and "initializations" show that this style would be considered an error in both.

**Pull request merged**